driver/format/RowBinaryWithNamesAndTypes.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include "driver/platform/platform.h"
+#include "driver/result_set.h"
+
+class RowBinaryWithNamesAndTypesResultSet
+    : public ResultSet
+{
+public:
+    explicit RowBinaryWithNamesAndTypesResultSet(std::istream & stream, std::unique_ptr<ResultMutator> && mutator);
+    virtual ~RowBinaryWithNamesAndTypesResultSet() override = default;
+
+protected:
+    virtual bool readNextRow(Row & row) override;
+
+private:
+    void readSize(std::uint64_t & dest);
+
+    void readValue(bool & dest);
+    void readValue(std::string & dest);
+    void readValue(std::string & dest, const std::uint64_t size);
+
+    template <typename T>
+    void readPOD(T & dest) {
+        constexpr auto size = sizeof(T);
+        raw_stream.read(reinterpret_cast<char *>(&dest), size);
+
+        if (raw_stream.gcount() != size)
+            throw std::runtime_error("Incomplete result received, expected size: " + std::to_string(size));
+    }
+
+    void readValue(Field & dest, ColumnInfo & column_info);
+
+    template <typename T>
+    void readValueAs(Field & dest, ColumnInfo & column_info) {
+        T value;
+        readValue(value, column_info);
+        dest.data = std::move(value);
+    }
+
+    void readValue(DataSourceType< DataSourceTypeId::Date        > & dest, ColumnInfo & column_info);
```

Comment:
IMO `DataSourceType< DataSourceTypeId::Date>` is a bit verbose, would you consider adding a type-aliases `DataSourceTypeDate`, etc ?

Reply:
I am not a big fan of aliases. You need to decompose those aliases in your mind each time you meet them, and store more "non-uniform" piece of data in your head, which spends your working memory resources.

I.e., while this may look like more raw data (in terms of characters):
```
DataSourceType< DataSourceTypeId::DateTime >
DataSourceType< DataSourceTypeId::Decimal   >
DataSourceType< DataSourceTypeId::Int64        >
```

this is more data in terms of unique slots in your memory:
```
DataSourceTypeDateTime
DataSourceTypeDecimal
DataSourceTypeInt64
```
(but the latter is better than using some random names, obviously.)

Reply:
In other words, there is a hidden structure in this naming: `DataSourceTypeDateTime`.

Reply:
Whereas, that stricture is explicit in: `DataSourceType< DataSourceTypeId::DateTime >`
So you don't need to do extra work.

Reply:
IMO, right now it looks "guts out": the 'hidden' struct here is a mere implementation detail.

Reply:
Oh, it's not an underlying struct, it's a type id enum value.